zeeguu/api/api/nlp.py

```diff
@@ -58,3 +59,28 @@ def annotate_clues():
 
     return json_result(updated_words)
 
+# ---------------------------------------------------------------------------
+@api.route("/get_sentences_for_wo", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_sentences_for_wo():
+    article_text = request.form.get("article_text", "")
+    bookmark_context = request.form.get("bookmark_context", "")
+    language = request.form.get("language")
+
+    if language not in SpacyWrappers.keys():
+        return "Language not supported"
+    
+    nlp_pipe = SpacyWrappers[language]
+    sentences = nlp_pipe.get_sent_list(article_text)
+    filtered_sentences = [sent for sent in sentences if len(nlp_pipe.tokenize_sentence(sent)) <= 15]
+    
+    context_doc = nlp_pipe.spacy_pipe(bookmark_context)
+    heap = []
+    for f_sent in filtered_sentences:
+        sent_doc = nlp_pipe.spacy_pipe(f_sent)
+        heapq.heappush(heap, (-context_doc.similarity(sent_doc), f_sent))
+
+    _, most_similar_sent = heap.pop()
```

Comment:
so you're throwing away the most similar sentence - i guess that's the bookmark sentence? to make it clear maybe simply do `_,_ = heap.pop()`

Reply:
There was actually a bug here. I was trying to put this change in order to allow us to check results directly in the console without having to push changes again. 

So I wanted to return the top sentence, but also the rest of the sentences in the article, I made it more clear now, and instead, I send a json with the top 1 and top 10 sentences (with the similarity).